applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/formfinding_strategy.hpp

```diff
@@ -0,0 +1,265 @@
+                return true;
+            }
+
+            void UpdateDatabase(
+                    TSystemMatrixType& A,
+                    TSystemVectorType& Dx,
+                    TSystemVectorType& b,
+                    const bool MoveMesh
+                ) override
+                {
+                    BaseType::UpdateDatabase(A,Dx, b, MoveMesh);
+                    for(auto& r_node : mrFormFindingModelPart.Nodes()){
+                        // Updating reference
+                        const array_1d<double, 3>& disp = r_node.FastGetSolutionStepValue(DISPLACEMENT);
+                        array_1d<double, 3>& disp_non_historical = r_node.GetValue(DISPLACEMENT);
+
+                        disp_non_historical = disp_non_historical + disp;
+                        r_node.GetInitialPosition() += disp;
+
+                        r_node.FastGetSolutionStepValue(DISPLACEMENT) = ZeroVector(3);
+                    }
+                    ProjectVectorOnSurfaceUtility::Execute(mrFormFindingModelPart, mProjectionSettings);
+
+                    PrintResults();
+                }
+
+                FormfindingStrategy(const FormfindingStrategy& Other) {};
+
+                void EchoInfo(const unsigned int IterationNumber) override
+                {
+                    BaseType::EchoInfo(IterationNumber);
+                    mIterationNumber = IterationNumber;
+                }
+
+                void PrintResults()
+                {
+                    if (mPrintingFormat=="all"){
+                        PrintVtkFiles(mIterationNumber);
+                        PrintGiDFiles(mIterationNumber);
+                    }
+                    else if (mPrintingFormat=="vtk") PrintVtkFiles(mIterationNumber);
+                    else if (mPrintingFormat=="gid") PrintGiDFiles(mIterationNumber);
+                    else;
+                }
+
+                void FinalizeSolutionStep() override
+                {
+                    BaseType::FinalizeSolutionStep();
+                    if (mWriteNewReferenceFile) WriteNewMdpaFile();
+                }
+
+                void WriteNewMdpaFile()
+                {
+                    Matrix output_matrix;
+                    const ProcessInfo temp_process_info;
+                    for(auto& r_element : mrFormFindingModelPart.Elements()){
+                        r_element.Calculate(MEMBRANE_PRESTRESS,output_matrix,temp_process_info);
+                        r_element.SetValue(MEMBRANE_PRESTRESS,output_matrix);
+                    }
+
+                    // write new mdpa
+                    // erase properties
+                    for (auto& prop: BaseType::GetModelPart().rProperties())
+                        prop.Data().Clear();
+
+
+                    ModelPartIO model_part_io("formfinding_result_model", IO::WRITE);
+                    model_part_io.WriteModelPart(BaseType::GetModelPart());
+                }
+
+                void PrintVtkFiles(const int& rIterationNumber)
```

Comment:
```suggestion
                void PrintVtkFiles(const int rIterationNumber)
```
reference is not needed for `const int`